File: A's/py/236A.py
def main() -> None:
    w: str = str(input())
    distinct: list[chr] = []

    for c in w:
        if c not in distinct:
            distinct.append(c)

    n: int = len(distinct)
    if (n % 2 == 0):  # Female
        print("CHAT WITH HER!")
    else:  # Male
        print("IGNORE HIM!")

if __name__ == "__main__":
    main()

File: A's/cpp/236A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    string w; cin >> w;

    vector<char> distinct;

    for (auto c : w) {
        if (find(distinct.begin(), distinct.end(), c) == distinct.end()) {
            distinct.push_back(c);
        }
    }

    int l = distinct.size();
    if (l % 2 == 0) {  // Female
        cout << "CHAT WITH HER!" << endl;
    }
    else {  // Male
        cout << "IGNORE HIM!" << endl;
    }
    
    return 0;
}